score-server/src/test/java/bio/overture/score/server/security/DownloadAuthorizationTest.java

```diff
@@ -0,0 +1,131 @@
+      PROJECT_SCOPE, "test.CODE.upload", "test2.download");
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertTrue(result);
+  }
+
+  @Test
+  public void test_authorization_ok_controlled_global_scope() {
+    val sut = getSut(CONTROLLED_ACCESS);
+
+    val authentication = getAuthentication(false, "test1.download", "test.OTHER-CODE.upload",
+      GLOBAL_SCOPE, "test.CODE.upload", "test2.download");
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertTrue(result);
+  }
+
+
+  @Test
+  public void test_authorization_ok_open_no_scopes() {
+    val sut = getSut(OPEN_ACCESS);
+
+    val authentication = getAuthentication(false);
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertTrue(result);
+  }
+
+  @Test
+  public void test_authorization_ok_open_no_scopes_expired_token() {
+    val sut = getSut(OPEN_ACCESS);
+
+    val authentication = getAuthentication(true);
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertTrue(result);
+  }
+
+  @Test
+  public void test_authorization_fail_controlled_no_matching_scopes() {
+    val sut = getSut(CONTROLLED_ACCESS);
+
+    val authentication = getAuthentication(false, "test1.download", "test.OTHER-CODE.upload",
+      "test.ALT-CODE.upload", "test.CODE.upload", "test2.download");
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertFalse(result);
+  }
+
+  @Test
+  public void test_authorization_fail_controlled_right_project_wrong_operation() {
+    val sut = getSut(CONTROLLED_ACCESS);
+    val authentication = getAuthentication(false, "test1.download", "test.OTHER-CODE.upload",
+      "test." + TEST_PROJECT + ".upload", "test.CODE.upload", "test2.download");
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertFalse(result);
+  }
+
+  @Test
+  public void test_authorization_fail_controlled_right_project_expiredToken() {
+    val sut = getSut(CONTROLLED_ACCESS);
+
+    val authentication = getAuthentication(true, "test1.download", "test.OTHER-CODE.upload",
+      PROJECT_SCOPE, "test.CODE.upload", "test2.download");
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertFalse(result);
+  }
+
+  @Test
+  public void test_authorization_fail_controlled_global_expiredToken() {
+    val sut = getSut(CONTROLLED_ACCESS);
+
+    val authentication = getAuthentication(true, "test1.download", "test.OTHER-CODE.upload",
+      GLOBAL_SCOPE, "test.CODE.upload", "test2.download");
+    val result = sut.authorize(authentication, TEST_OBJECT_ID);
+    assertFalse(result);
+  }
+
+  private Authentication getAuthentication(boolean isExpired, String... scopes) {
+    val request = new OAuth2Request(emptyMap(), "", emptyList(), true,
+      new TreeSet<>(Arrays.asList(scopes)), emptySet(), "", emptySet(), emptyMap());
+
+    return ExpiringOauth2Authentication.from(new OAuth2Authentication(request, null),
+      isExpired ? 0 : 3600);
+  }
```

Comment:
instead of all of this, can you do something similar to this:
```java
val map = new HashMap<String, Object>();
map.put("exp", isExpired ? 0 : 3600);
return new AccessTokenConverterWithExpiry().extractAuthentication(map);
```